axlearn/common/attention.py

```diff
@@ -3874,6 +3874,7 @@ def build_remat_spec(
     ],
     self_attention: bool = True,
     feed_forward: bool = False,
+    offload: bool = False,
```

Comment:
Instead of a bool, should we allow the caller to customize `offload_dst` directly?

```suggestion
    offload_dst: Optional[Literal["pinned_host"]] = None,
```

This will make the API more extensible and closer to the JAX API.

Reply:
Agree. I will take another stab at this PR with focus on staying closer to JAX API and extensibility.

Reply:
Resolved, could you review again?